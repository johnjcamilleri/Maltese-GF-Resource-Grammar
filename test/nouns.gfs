-- Treebank nouns test
-- Run using the following command:
-- ./_gf < test/nouns.gfs > test/nouns.out

se utf8
i AllMlt.gf
rf -lines -tree -file=test/nouns.trees | l -table
