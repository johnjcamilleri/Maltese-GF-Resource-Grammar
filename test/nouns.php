#!/usr/bin/php
<?php

// Timing
$time_start = microtime(true);

// Just some variables
$outfile = 'test/nouns.out';
$command = './_gf < test/nouns.gfs > '.$outfile;
$htmlfile = 'test/nouns.out.html';
$TEXT = '';

// ================

// Execute GF stuff and capture output
if (@$GLOBALS['argv'][1] == '--cached') {
	echo " Reading directly from {$outfile}\n";
} else {
	echo " Running GF\n";
	chdir( dirname(__FILE__) . '/../' );
	exec($command, $out, $return_status);
	if ($return_status != 0)
		die (" Failed.\n");
}
$output = file($outfile, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
if ($output === false)
		die (" Failed reading {$outfile}.\n");

// ================

// Recursive function for nesting as deep as we need
function nest($stack, &$branch) {
	if (count($stack) > 1) {
		$key = array_shift($stack);
		nest($stack, $branch[$key]);
	} else {
		$branch = $stack[0];
	}
}

// Process line by line & put into heirarchical array
$tree = array(); // overall structure
$ix = 0; // item index
$console_output = '';
foreach ($output as $n => $line) {

	// Skip timings
	if (preg_match('/^(> )?\d+ msec/', $line)) continue;

	// Process items
	elseif (preg_match('/^(> )?(\w) \. (.*)/', $line, $matches)) {

		// Have we reached a new entry?
		if ($matches[1]) {
			$ix++;
		}

		// Split at arrow and nest
		$columns = explode('=>',$matches[3]);
		$columns = array_map('trim', $columns);
		nest($columns, $tree[$ix][$matches[2]]);
	}

	// Anything else is compiler output
	else
		$console_output .= $line . "\n";
}

// ================

// Process recursively for output

function disp($branch, $depth=0) {
	if (is_array($branch)) {
		foreach ($branch as $k => $v) {
			//printf("\n%s%-15s", str_repeat("\t", $depth), $k);
			echo "<div><b>$k</b>";
			disp($v, $depth+1);
			echo "</div>";
		}
	} else {
		echo $branch == '[]' ? '' : "<em>$branch</em>";
	}
}
/*
function dispHTML($branch, $depth=0) {
	if (is_array($branch)) {
		foreach ($branch as $k => $v) {
			printf('<h%2$d>%1$s</h%2$d>'."\n", $k, $depth+1);
			dispHTML($v, $depth+1);
		}
	} else {
		echo "<p>$branch</p>\n";
	}
}
*/

ob_start();
foreach ($tree as $item) {
	disp($item);
	echo "<hr/>";
}
$TEXT_out = ob_get_contents();
ob_end_clean();


// ================

// Format into HTML
echo " Formatting to HTML\n";
$HTML = <<<HTML
<html>
<head>
	<meta http-equiv="Content-type" content="text/html;charset=UTF-8">
	<style type="text/css">
		body { font: 18px/1.5 serif; color:#999; }
		div { margin-left: 2em; }
		#text-output { margin:0; }
		#text-output b { display: inline-block; width: 10em; font-weight:normal; }
		#text-output em { font-style: italic; color:#000; }

		table { border-collapse:collapse; border-spacing:0; }
		tr:nth-child(odd) { background-color:#eee; }
		tr.separator { background-color:#000; font-size:2px; }
		td { padding:0.2em 1em; border-width:1px 0; border-style:solid; border-color:#ccc; }
		td:last-child { font-style:italic; }
	</style>
</head>
<body>


<div id="text-output">
{$TEXT_out}
</div>

</body>
</html>
HTML;


/*
<table>{$HTML_table}</table>
<a href="#compiler-output">Compiler output</a>
<pre id="compiler-output">{$HTML_console}</pre>
*/


// Save to file
echo " Writing to file $htmlfile\n";
@unlink($htmlfile);
file_put_contents($htmlfile, $HTML);

// ================

// Stop timing
$time_end = microtime(true);
$time_taken = $time_end - $time_start;
printf(" Done in %.2fs (completed at %s)\n", $time_taken, date('H:i'));
