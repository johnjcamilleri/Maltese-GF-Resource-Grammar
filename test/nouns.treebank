airplane_N
animal_N
apartment_N
apple_N
art_N
ashes_N
baby_N
back_N
bank_N
bark_N
beer_N
belly_N
bike_N
bird_N
blood_N
boat_N
bone_N
book_N
boot_N
boss_N
boy_N
bread_N
breast_N
brother_N2
butter_N
camera_N
cap_N
car_N
carpet_N
cat_N
ceiling_N
chair_N
cheese_N
child_N
church_N
city_N
cloud_N
coat_N
computer_N
country_N
cousin_N
cow_N
day_N
distance_N3
doctor_N
dog_N
door_N
dust_N
ear_N
earth_N
egg_N
enemy_N
eye_N
factory_N
fat_N
father_N2
feather_N
fingernail_N
fire_N
fish_N
floor_N
flower_N
fog_N
foot_N
forest_N
fridge_N
friend_N
fruit_N
garden_N
girl_N
glove_N
gold_N
grammar_N
grass_N
guts_N
hair_N
hand_N
harbour_N
hat_N
head_N
heart_N
hill_N
horn_N
horse_N
house_N
husband_N
ice_N
industry_N
iron_N
king_N
knee_N
lake_N
lamp_N
language_N
leaf_N
leather_N
leg_N
liver_N
louse_N
love_N
man_N
meat_N
milk_N
moon_N
mother_N2
mountain_N
mouth_N
music_N
name_N
neck_N
newspaper_N
night_N
nose_N
number_N
oil_N
paper_N
peace_N
pen_N
person_N
planet_N
plastic_N
policeman_N
priest_N
queen_N
question_N
radio_N
rain_N
reason_N
religion_N
restaurant_N
river_N
road_N
rock_N
roof_N
root_N
rope_N
rubber_N
rule_N
salt_N
sand_N
school_N
science_N
sea_N
seed_N
sheep_N
ship_N
shirt_N
shoe_N
shop_N
silver_N
sister_N
skin_N
sky_N
smoke_N
snake_N
snow_N
sock_N
song_N
star_N
steel_N
stick_N
stone_N
stove_N
student_N
sun_N
table_N
tail_N
teacher_N
television_N
tongue_N
tooth_N
train_N
tree_N
university_N
village_N
war_N
water_N
wife_N
wind_N
window_N
wine_N
wing_N
woman_N
wood_N
worm_N
year_N
