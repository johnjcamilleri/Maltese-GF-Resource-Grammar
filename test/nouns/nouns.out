s (Singular Singulative) : ajruplan
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ajruplani
s (Plural Indeterminate) :


s (Singular Singulative) : annimal
s (Singular Collective) :
s Dual :
s (Plural Determinate) : annimali
s (Plural Indeterminate) :


s (Singular Singulative) : appartament
s (Singular Collective) :
s Dual :
s (Plural Determinate) : appartamenti
s (Plural Indeterminate) :


s (Singular Singulative) : tuffieħa
s (Singular Collective) : tuffieħ
s Dual :
s (Plural Determinate) : tuffieħiet
s (Plural Indeterminate) :


s (Singular Singulative) : arti
s (Singular Collective) :
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : tarbija
s (Singular Collective) :
s Dual :
s (Plural Determinate) : trabi
s (Plural Indeterminate) :


s (Singular Singulative) : dahar
s (Singular Collective) :
s Dual :
s (Plural Determinate) : dhur
s (Plural Indeterminate) :


s (Singular Singulative) : bank
s (Singular Collective) :
s Dual :
s (Plural Determinate) : bankijiet
s (Plural Indeterminate) :


s (Singular Singulative) : qoxra
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qoxriet
s (Plural Indeterminate) :


s (Singular Singulative) : birra
s (Singular Collective) :
s Dual :
s (Plural Determinate) : birer
s (Plural Indeterminate) :


s (Singular Singulative) : żaqq
s (Singular Collective) :
s Dual :
s (Plural Determinate) : żquq
s (Plural Indeterminate) :


s (Singular Singulative) : rota
s (Singular Collective) :
s Dual :
s (Plural Determinate) : roti
s (Plural Indeterminate) :


s (Singular Singulative) : għasfur
s (Singular Collective) :
s Dual :
s (Plural Determinate) : għasafar
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : demm
s Dual :
s (Plural Determinate) : dmija
s (Plural Indeterminate) :


s (Singular Singulative) : dgħajsa
s (Singular Collective) :
s Dual :
s (Plural Determinate) : dgħajjes
s (Plural Indeterminate) :


s (Singular Singulative) : għadma
s (Singular Collective) : għadam
s Dual :
s (Plural Determinate) : għadmiet
s (Plural Indeterminate) :


s (Singular Singulative) : ktieb
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kotba
s (Plural Indeterminate) :


s (Singular Singulative) : żarbun
s (Singular Collective) :
s Dual :
s (Plural Determinate) : żraben
s (Plural Indeterminate) :


s (Singular Singulative) : mgħallem
s (Singular Collective) :
s Dual :
s (Plural Determinate) : mgħallmin
s (Plural Indeterminate) :


s (Singular Singulative) : tifel
s (Singular Collective) :
s Dual :
s (Plural Determinate) : tfal
s (Plural Indeterminate) :


s (Singular Singulative) : ħobża
s (Singular Collective) : ħobż
s Dual :
s (Plural Determinate) : ħobżiet
s (Plural Indeterminate) :


s (Singular Singulative) : sider
s (Singular Collective) :
s Dual :
s (Plural Determinate) : sdur
s (Plural Indeterminate) :


s (Singular Singulative) : ħu
s (Singular Collective) :
s Dual :
s (Plural Determinate) : aħwa
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : butir
s Dual :
s (Plural Determinate) : butirijiet
s (Plural Indeterminate) :


s (Singular Singulative) : kamera
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kameras
s (Plural Indeterminate) :


s (Singular Singulative) : beritta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : beritti
s (Plural Indeterminate) :


s (Singular Singulative) : karozza
s (Singular Collective) :
s Dual :
s (Plural Determinate) : karozzi
s (Plural Indeterminate) :


s (Singular Singulative) : tapit
s (Singular Collective) :
s Dual :
s (Plural Determinate) : tapiti
s (Plural Indeterminate) :


s (Singular Singulative) : qattus
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qtates
s (Plural Indeterminate) :


s (Singular Singulative) : saqaf
s (Singular Collective) :
s Dual :
s (Plural Determinate) : soqfa
s (Plural Indeterminate) :


s (Singular Singulative) : siġġu
s (Singular Collective) :
s Dual :
s (Plural Determinate) : siġġijiet
s (Plural Indeterminate) :


s (Singular Singulative) : ġobna
s (Singular Collective) : ġobon
s Dual :
s (Plural Determinate) : ġobniet
s (Plural Indeterminate) :




s (Singular Singulative) : knisja
s (Singular Collective) :
s Dual :
s (Plural Determinate) : knejjes
s (Plural Indeterminate) :


s (Singular Singulative) : belt
s (Singular Collective) :
s Dual :
s (Plural Determinate) : bliet
s (Plural Indeterminate) :


s (Singular Singulative) : sħaba
s (Singular Collective) : sħab
s Dual :
s (Plural Determinate) : sħabiet
s (Plural Indeterminate) :


s (Singular Singulative) : kowt
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kowtijiet
s (Plural Indeterminate) :


s (Singular Singulative) : kompjuter
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kompjuters
s (Plural Indeterminate) :


s (Singular Singulative) : pajjiż
s (Singular Collective) :
s Dual :
s (Plural Determinate) : pajjiżi
s (Plural Indeterminate) :


s (Singular Singulative) : kuġin
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kuġini
s (Plural Indeterminate) :


s (Singular Singulative) : baqra
s (Singular Collective) : baqar
s Dual : baqartejn
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : ġurnata
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ġranet
s (Plural Indeterminate) :


s (Singular Singulative) : distanza
s (Singular Collective) :
s Dual :
s (Plural Determinate) : distanzi
s (Plural Indeterminate) :


s (Singular Singulative) : tabib
s (Singular Collective) :
s Dual :
s (Plural Determinate) : tobba
s (Plural Indeterminate) :


s (Singular Singulative) : kelb
s (Singular Collective) :
s Dual :
s (Plural Determinate) : klieb
s (Plural Indeterminate) :


s (Singular Singulative) : bieb
s (Singular Collective) :
s Dual :
s (Plural Determinate) : bibien
s (Plural Indeterminate) :


s (Singular Singulative) : traba
s (Singular Collective) : trab
s Dual :
s (Plural Determinate) : trabiet
s (Plural Indeterminate) :


s (Singular Singulative) : widna
s (Singular Collective) :
s Dual : widnejn
s (Plural Determinate) : widniet
s (Plural Indeterminate) :


s (Singular Singulative) : art
s (Singular Collective) :
s Dual :
s (Plural Determinate) : artijiet
s (Plural Indeterminate) :


s (Singular Singulative) : bajda
s (Singular Collective) : bajd
s Dual :
s (Plural Determinate) : bajdiet
s (Plural Indeterminate) :


s (Singular Singulative) : għadu
s (Singular Collective) :
s Dual :
s (Plural Determinate) : għedewwa
s (Plural Indeterminate) :


s (Singular Singulative) : għajn
s (Singular Collective) :
s Dual : għajnejn
s (Plural Determinate) : għajnejn
s (Plural Indeterminate) : għejun


s (Singular Singulative) : fabbrika
s (Singular Collective) :
s Dual :
s (Plural Determinate) : fabbriki
s (Plural Indeterminate) :


s (Singular Singulative) : xaħma
s (Singular Collective) : xaħam
s Dual :
s (Plural Determinate) : xaħmiet
s (Plural Indeterminate) :


s (Singular Singulative) : missier
s (Singular Collective) :
s Dual :
s (Plural Determinate) : missierijiet
s (Plural Indeterminate) :


s (Singular Singulative) : rixa
s (Singular Collective) : rix
s Dual :
s (Plural Determinate) : rixiet
s (Plural Indeterminate) :


s (Singular Singulative) : difer
s (Singular Collective) :
s Dual : difrejn
s (Plural Determinate) : dwiefer
s (Plural Indeterminate) :


s (Singular Singulative) : nar
s (Singular Collective) :
s Dual :
s (Plural Determinate) : nirien
s (Plural Indeterminate) :


s (Singular Singulative) : ħuta
s (Singular Collective) : ħut
s Dual :
s (Plural Determinate) : ħutiet
s (Plural Indeterminate) :




s (Singular Singulative) : fjura
s (Singular Collective) :
s Dual :
s (Plural Determinate) : fjuri
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : ċpar
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : sieq
s (Singular Collective) :
s Dual : saqajn
s (Plural Determinate) : saqajn
s (Plural Indeterminate) :


s (Singular Singulative) : foresta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : foresti
s (Plural Indeterminate) :


s (Singular Singulative) : friġġ
s (Singular Collective) :
s Dual :
s (Plural Determinate) : friġġijiet
s (Plural Indeterminate) :


s (Singular Singulative) : ħabib
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ħbieb
s (Plural Indeterminate) :


s (Singular Singulative) : frotta
s (Singular Collective) : frott
s Dual :
s (Plural Determinate) : frottiet
s (Plural Indeterminate) :


s (Singular Singulative) : ġnien
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ġonna
s (Plural Indeterminate) :


s (Singular Singulative) : tifla
s (Singular Collective) :
s Dual :
s (Plural Determinate) : tfal
s (Plural Indeterminate) :


s (Singular Singulative) : ingwanta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ingwanti
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : deheb
s Dual :
s (Plural Determinate) : dehbijiet
s (Plural Indeterminate) :


s (Singular Singulative) : grammatika
s (Singular Collective) :
s Dual :
s (Plural Determinate) : grammatiki
s (Plural Indeterminate) :


s (Singular Singulative) : ħaxixa
s (Singular Collective) : ħaxix
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) : ħxejjex


s (Singular Singulative) : musrana
s (Singular Collective) :
s Dual :
s (Plural Determinate) : musraniet
s (Plural Indeterminate) : msaren


s (Singular Singulative) : xagħar
s (Singular Collective) :
s Dual :
s (Plural Determinate) : xagħariet
s (Plural Indeterminate) : xgħur


s (Singular Singulative) : id
s (Singular Collective) :
s Dual : idejn
s (Plural Determinate) : idejn
s (Plural Indeterminate) :


s (Singular Singulative) : port
s (Singular Collective) :
s Dual :
s (Plural Determinate) : portijiet
s (Plural Indeterminate) :


s (Singular Singulative) : kappell
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kpiepel
s (Plural Indeterminate) :


s (Singular Singulative) : ras
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rjus
s (Plural Indeterminate) :


s (Singular Singulative) : qalb
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qlub
s (Plural Indeterminate) :


s (Singular Singulative) : għolja
s (Singular Collective) :
s Dual :
s (Plural Determinate) : għoljiet
s (Plural Indeterminate) :


s (Singular Singulative) : ħorn
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ħornijiet
s (Plural Indeterminate) :


s (Singular Singulative) : żiemel
s (Singular Collective) :
s Dual :
s (Plural Determinate) : żwiemel
s (Plural Indeterminate) :


s (Singular Singulative) : dar
s (Singular Collective) :
s Dual :
s (Plural Determinate) : djar
s (Plural Indeterminate) :


s (Singular Singulative) : raġel
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rġiel
s (Plural Indeterminate) :


s (Singular Singulative) : silġ
s (Singular Collective) :
s Dual :
s (Plural Determinate) : silġiet
s (Plural Indeterminate) :


s (Singular Singulative) : industrija
s (Singular Collective) :
s Dual :
s (Plural Determinate) : industriji
s (Plural Indeterminate) :


s (Singular Singulative) : ħadida
s (Singular Collective) : ħadid
s Dual :
s (Plural Determinate) : ħadidiet
s (Plural Indeterminate) : ħdejjed


s (Singular Singulative) : re
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rejjiet
s (Plural Indeterminate) :


s (Singular Singulative) : rkoppa
s (Singular Collective) :
s Dual : rkopptejn
s (Plural Determinate) : rkoppiet
s (Plural Indeterminate) :


s (Singular Singulative) : għadira
s (Singular Collective) :
s Dual :
s (Plural Determinate) : għadajjar
s (Plural Indeterminate) :


s (Singular Singulative) : lampa
s (Singular Collective) :
s Dual :
s (Plural Determinate) : lampi
s (Plural Indeterminate) :


s (Singular Singulative) : lingwa
s (Singular Collective) :
s Dual :
s (Plural Determinate) : lingwi
s (Plural Indeterminate) :


s (Singular Singulative) : werqa
s (Singular Collective) : weraq
s Dual : werqtejn
s (Plural Determinate) : werqiet
s (Plural Indeterminate) :


s (Singular Singulative) : ġilda
s (Singular Collective) : ġild
s Dual :
s (Plural Determinate) : ġildiet
s (Plural Indeterminate) : ġlud


s (Singular Singulative) : riġel
s (Singular Collective) :
s Dual : riġlejn
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : fwied
s (Singular Collective) :
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) : ifdwa


s (Singular Singulative) : qamla
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qamliet
s (Plural Indeterminate) :


s (Singular Singulative) : mħabba
s (Singular Collective) :
s Dual :
s (Plural Determinate) : mħabbiet
s (Plural Indeterminate) :


s (Singular Singulative) : raġel
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rġiel
s (Plural Indeterminate) :


s (Singular Singulative) : laħam
s (Singular Collective) :
s Dual :
s (Plural Determinate) : laħmiet
s (Plural Indeterminate) : laħmijiet


s (Singular Singulative) :
s (Singular Collective) : ħalib
s Dual :
s (Plural Determinate) : ħalibijiet
s (Plural Indeterminate) : ħlejjeb


s (Singular Singulative) : qamar
s (Singular Collective) :
s Dual :
s (Plural Determinate) : oqmra
s (Plural Indeterminate) :


s (Singular Singulative) : omm
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ommijiet
s (Plural Indeterminate) :


s (Singular Singulative) : muntanja
s (Singular Collective) :
s Dual :
s (Plural Determinate) : muntanji
s (Plural Indeterminate) :


s (Singular Singulative) : ħalq
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ħluq
s (Plural Indeterminate) :


s (Singular Singulative) : musika
s (Singular Collective) :
s Dual :
s (Plural Determinate) : musiki
s (Plural Indeterminate) :


s (Singular Singulative) : isem
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ismijiet
s (Plural Indeterminate) :


s (Singular Singulative) : għonq
s (Singular Collective) :
s Dual :
s (Plural Determinate) : għenuq
s (Plural Indeterminate) :


s (Singular Singulative) : gazzetta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : gazzetti
s (Plural Indeterminate) :


s (Singular Singulative) : lejl
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ljieli
s (Plural Indeterminate) :


s (Singular Singulative) : mnieħer
s (Singular Collective) :
s Dual :
s (Plural Determinate) : mniħrijiet
s (Plural Indeterminate) :


s (Singular Singulative) : numru
s (Singular Collective) :
s Dual :
s (Plural Determinate) : numri
s (Plural Indeterminate) :


s (Singular Singulative) : żejt
s (Singular Collective) :
s Dual :
s (Plural Determinate) : żjut
s (Plural Indeterminate) :


s (Singular Singulative) : karta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : karti
s (Plural Indeterminate) :


s (Singular Singulative) : paċi
s (Singular Collective) :
s Dual :
s (Plural Determinate) : paċijiet
s (Plural Indeterminate) :


s (Singular Singulative) : pinna
s (Singular Collective) :
s Dual :
s (Plural Determinate) : pinen
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : persuna
s Dual :
s (Plural Determinate) : persuni
s (Plural Indeterminate) :


s (Singular Singulative) : pjaneta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : pjaneti
s (Plural Indeterminate) :


s (Singular Singulative) : plastik
s (Singular Collective) :
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : pulizija
s (Singular Collective) :
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : qassis
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qassisin
s (Plural Indeterminate) :


s (Singular Singulative) : reġina
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rġejjen
s (Plural Indeterminate) :


s (Singular Singulative) : mistoqsija
s (Singular Collective) :
s Dual :
s (Plural Determinate) : mistoqsijiet
s (Plural Indeterminate) :


s (Singular Singulative) : radju
s (Singular Collective) :
s Dual :
s (Plural Determinate) : radjijiet
s (Plural Indeterminate) :


s (Singular Singulative) : xita
s (Singular Collective) :
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : raġun
s (Singular Collective) :
s Dual :
s (Plural Determinate) : raġunijiet
s (Plural Indeterminate) :


s (Singular Singulative) : reliġjon
s (Singular Collective) :
s Dual :
s (Plural Determinate) : reliġjonijiet
s (Plural Indeterminate) :


s (Singular Singulative) : restorant
s (Singular Collective) :
s Dual :
s (Plural Determinate) : restoranti
s (Plural Indeterminate) :


s (Singular Singulative) : xmara
s (Singular Collective) :
s Dual :
s (Plural Determinate) : xmajjar
s (Plural Indeterminate) :


s (Singular Singulative) : triq
s (Singular Collective) :
s Dual :
s (Plural Determinate) : triqat
s (Plural Indeterminate) : toroq


s (Singular Singulative) : blata
s (Singular Collective) : blat
s Dual :
s (Plural Determinate) : blatiet
s (Plural Indeterminate) : blajjiet


s (Singular Singulative) : saqaf
s (Singular Collective) :
s Dual :
s (Plural Determinate) : soqfa
s (Plural Indeterminate) :


s (Singular Singulative) : qħerq
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qħeruq
s (Plural Indeterminate) :


s (Singular Singulative) : ħabel
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ħbula
s (Plural Indeterminate) :


s (Singular Singulative) : gomma
s (Singular Collective) :
s Dual :
s (Plural Determinate) : gomom
s (Plural Indeterminate) :


s (Singular Singulative) : regola
s (Singular Collective) :
s Dual :
s (Plural Determinate) : regoli
s (Plural Indeterminate) :


s (Singular Singulative) : melħ
s (Singular Collective) :
s Dual :
s (Plural Determinate) : melħiet
s (Plural Indeterminate) :


s (Singular Singulative) : ramla
s (Singular Collective) : ramel
s Dual :
s (Plural Determinate) : ramliet
s (Plural Indeterminate) : rmiel


s (Singular Singulative) : skola
s (Singular Collective) :
s Dual :
s (Plural Determinate) : skejjel
s (Plural Indeterminate) :


s (Singular Singulative) : xjenza
s (Singular Collective) :
s Dual :
s (Plural Determinate) : xjenzi
s (Plural Indeterminate) :


s (Singular Singulative) : baħar
s (Singular Collective) :
s Dual : baħrejn
s (Plural Determinate) : ibħra
s (Plural Indeterminate) :


s (Singular Singulative) : żerriegħa
s (Singular Collective) :
s Dual :
s (Plural Determinate) : żerrigħat
s (Plural Indeterminate) :


s (Singular Singulative) : nagħġa
s (Singular Collective) : nagħaġ
s Dual :
s (Plural Determinate) : nagħġiet
s (Plural Indeterminate) :


s (Singular Singulative) : vapur
s (Singular Collective) :
s Dual :
s (Plural Determinate) : vapuri
s (Plural Indeterminate) :


s (Singular Singulative) : qmis
s (Singular Collective) :
s Dual :
s (Plural Determinate) : qomos
s (Plural Indeterminate) :


s (Singular Singulative) : żarbun
s (Singular Collective) :
s Dual :
s (Plural Determinate) : żraben
s (Plural Indeterminate) :


s (Singular Singulative) : ħanut
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ħwienet
s (Plural Indeterminate) :


s (Singular Singulative) : fidda
s (Singular Collective) :
s Dual :
s (Plural Determinate) : fided
s (Plural Indeterminate) :


s (Singular Singulative) : oħt
s (Singular Collective) :
s Dual :
s (Plural Determinate) : aħwa
s (Plural Indeterminate) :


s (Singular Singulative) : ġilda
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ġildiet
s (Plural Indeterminate) :


s (Singular Singulative) : sema
s (Singular Collective) :
s Dual :
s (Plural Determinate) : smewwiet
s (Plural Indeterminate) :


s (Singular Singulative) : duħħan
s (Singular Collective) :
s Dual :
s (Plural Determinate) : dħaħen
s (Plural Indeterminate) :


s (Singular Singulative) : serp
s (Singular Collective) :
s Dual :
s (Plural Determinate) : sriep
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : borra
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : kalzetta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kalzetti
s (Plural Indeterminate) :


s (Singular Singulative) : kanzunetta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kanzunetti
s (Plural Indeterminate) :


s (Singular Singulative) : stilla
s (Singular Collective) :
s Dual :
s (Plural Determinate) : stilel
s (Plural Indeterminate) :


s (Singular Singulative) : azzar
s (Singular Collective) :
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) :


s (Singular Singulative) : lasta
s (Singular Collective) :
s Dual :
s (Plural Determinate) : lasti
s (Plural Indeterminate) :


s (Singular Singulative) : ġebla
s (Singular Collective) : ġebel
s Dual :
s (Plural Determinate) : ġebliet
s (Plural Indeterminate) : ġbiel


s (Singular Singulative) : kuker
s (Singular Collective) :
s Dual :
s (Plural Determinate) : kukers
s (Plural Indeterminate) :


s (Singular Singulative) : student
s (Singular Collective) :
s Dual :
s (Plural Determinate) : studenti
s (Plural Indeterminate) :


s (Singular Singulative) : xemx
s (Singular Collective) :
s Dual :
s (Plural Determinate) : xmux
s (Plural Indeterminate) :


s (Singular Singulative) : mejda
s (Singular Collective) :
s Dual :
s (Plural Determinate) : mwejjed
s (Plural Indeterminate) :


s (Singular Singulative) : denb
s (Singular Collective) :
s Dual :
s (Plural Determinate) : dnieb
s (Plural Indeterminate) :


s (Singular Singulative) : għalliem
s (Singular Collective) :
s Dual :
s (Plural Determinate) : għalliema
s (Plural Indeterminate) :


s (Singular Singulative) : televixin
s (Singular Collective) :
s Dual :
s (Plural Determinate) : televixins
s (Plural Indeterminate) :


s (Singular Singulative) : lsien
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ilsna
s (Plural Indeterminate) :


s (Singular Singulative) : sinna
s (Singular Collective) :
s Dual :
s (Plural Determinate) : sinniet
s (Plural Indeterminate) : snien


s (Singular Singulative) : ferrovija
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ferroviji
s (Plural Indeterminate) :


s (Singular Singulative) : siġra
s (Singular Collective) : siġar
s Dual :
s (Plural Determinate) : siġriet
s (Plural Indeterminate) :


s (Singular Singulative) : università
s (Singular Collective) :
s Dual :
s (Plural Determinate) : universitàjiet
s (Plural Indeterminate) :


s (Singular Singulative) : raħal
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rħula
s (Plural Indeterminate) :


s (Singular Singulative) : gwerra
s (Singular Collective) :
s Dual :
s (Plural Determinate) : gwerrer
s (Plural Indeterminate) :


s (Singular Singulative) : ilma
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ilmijiet
s (Plural Indeterminate) :


s (Singular Singulative) : mara
s (Singular Collective) :
s Dual :
s (Plural Determinate) : nisa
s (Plural Indeterminate) :


s (Singular Singulative) : riħ
s (Singular Collective) :
s Dual :
s (Plural Determinate) : rjieħ
s (Plural Indeterminate) : rjiħat


s (Singular Singulative) : tieqa
s (Singular Collective) :
s Dual :
s (Plural Determinate) : twieqi
s (Plural Indeterminate) :


s (Singular Singulative) :
s (Singular Collective) : nbid
s Dual :
s (Plural Determinate) :
s (Plural Indeterminate) : nbejjed


s (Singular Singulative) : ġewnaħ
s (Singular Collective) :
s Dual :
s (Plural Determinate) : ġwienaħ
s (Plural Indeterminate) :


s (Singular Singulative) : mara
s (Singular Collective) :
s Dual :
s (Plural Determinate) : nisa
s (Plural Indeterminate) :


s (Singular Singulative) : injam
s (Singular Collective) :
s Dual :
s (Plural Determinate) : injamiet
s (Plural Indeterminate) :


s (Singular Singulative) : dudu
s (Singular Collective) : dud
s Dual :
s (Plural Determinate) : dudiet
s (Plural Indeterminate) : dwied


s (Singular Singulative) : sena
s (Singular Collective) :
s Dual : sentejn
s (Plural Determinate) : snin
s (Plural Indeterminate) :



12 msec
AllMltAbs> See you.
0 msec